executor/simple.go

```diff
@@ -230,18 +230,40 @@ func (e *SimpleExec) executeDropUser(s *ast.DropUserStmt) error {
 			}
 			continue
 		}
+
+		// begin a transaction to delete a user.
+		if _, err := e.ctx.(sqlexec.SQLExecutor).Execute(context.Background(), "begin"); err != nil {
+			fmt.Println(err)
+			return errors.Trace(err)
+		}
 		sql := fmt.Sprintf(`DELETE FROM %s.%s WHERE Host = "%s" and User = "%s";`, mysql.SystemDB, mysql.UserTable, user.Hostname, user.Username)
-		_, _, err = e.ctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQL(e.ctx, sql)
-		if err != nil {
+		if _, err := e.ctx.(sqlexec.SQLExecutor).Execute(context.Background(), sql); err != nil {
+			failedUsers = append(failedUsers, user.String())
```

Comment:
It's unlikely to meet a error here.

Reply:
Why? For example all the tikv nodes are down.

Reply:
Before commit, all the operation is cached in union store @shenli

Reply:
Delete statement need to get data from tikv first.

Reply:
If the err is not nil, we should return error.